R/bs-dependencies.R

```diff
@@ -116,14 +116,34 @@ bs_dependencies <- function(
         src = out_file_dir,
         stylesheet = basename(out_file),
         script = basename(js_files),
-        all_files = TRUE, # js_map_files
+        all_files = TRUE, # include font and map files
         meta = list(viewport = "width=device-width, initial-scale=1, shrink-to-fit=no")
       )
     ),
     theme$html_deps
   )
 }
 
+#' @rdname bs_dependencies
+#' @param func a function that takes a [bs_theme()] object (or `theme_default`) as input and
+#' and returns an [htmlDependency()] (or a list of them).
+#' @param theme_default In the case that `func` wants to be executed outside of a Shiny
+#' runtime (e.g., in an R Markdown doc), an expression that yields a value to provide to `func`.
+#' @param env environment for evaluation the expression provided to `theme_default`.
+#' @export
+bs_runtime_dependencies <- function(func, theme_default = bs_global_get(), envir = parent.env()) {
+  theme_default <- substitute(theme_default)
+  tagFunction(function() {
+    if (is_available("shiny") && shiny::isRunning()) {
```

Comment:
It may be better to check if shiny is loaded -- otherwise, this code will cause shiny to be loaded even if this is being used for an rmarkdown document.